Library/Homebrew/utils/tty.rb

```diff
@@ -0,0 +1,52 @@
+module Tty
+  module_function
+
+  def strip_ansi(string)
+    string.gsub(/\033\[\d+(;\d+)*m/, "")
+  end
+
+  def width
+    `/usr/bin/tput cols`.strip.to_i
+  end
+
+  def truncate(string)
+    (w = width).zero? ? string.to_s : string.to_s[0, w - 4]
+  end
+
+  COLOR_CODES = {
+    :red     => 31,
+    :green   => 32,
+    :yellow  => 33,
+    :blue    => 34,
+    :magenta => 35,
+    :cyan    => 36,
+    :default => 39,
+  }
+
+  STYLE_CODES = {
+    :reset         =>  0,
+    :bold          =>  1,
+    :italic        =>  3,
+    :underline     =>  4,
+    :strikethrough =>  9,
+    :no_underline  => 24,
+  }
+
+  CODES = COLOR_CODES.merge(STYLE_CODES)
+
+  CODES.each do |name, code|
+    define_singleton_method(name) do
+      @sequence ||= []
+      @sequence << code
+      self
+    end
+  end
+
+  def to_s
+    return "" if @sequence.nil?
+    return "" unless $stdout.tty?
+    "\033[#{@sequence.join(";")}m"
+  ensure
+    @sequence = nil
```

Comment:
I have to agree with Mike that I find the splatting and collecting a bit harder to follow. A possibly easier to read and still succinct alternative for the check could be:

```ruby
return "" if (@sequence || []).empty?
```